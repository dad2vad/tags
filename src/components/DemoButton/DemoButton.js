import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon';
import Link from '../Link';

class DemoButton extends PureComponent {
  render() {
    return (
      <Link url={this.props.url} className="button is-fixed is-small is-info is-outlined">
        <Icon name="eye" />
        <span>Demo</span>
      </Link>
    );
  }
}

DemoButton.propTypes = {
  url: PropTypes.string.isRequired,
};

DemoButton.defaultProps = {
  url: '',
};

export default DemoButton;
