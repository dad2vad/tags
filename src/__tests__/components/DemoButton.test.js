import React from 'react';
import { shallow } from 'enzyme';
import DemoButton from '../../components/DemoButton';

describe('DemoButton', () => {
  it('renders correctly', () => {
    const component = shallow(<DemoButton url="https://frameworks.now.sh" />);
    expect(component).toMatchSnapshot();
  });
});
