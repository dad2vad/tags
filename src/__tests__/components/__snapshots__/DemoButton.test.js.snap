// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoButton renders correctly 1`] = `
<Link
  className="button is-fixed is-small is-info is-outlined"
  url="https://frameworks.now.sh"
>
  <Icon
    name="eye"
  />
  <span>
    Demo
  </span>
</Link>
`;
